tests/manage/z_cluster/cluster_expansion/test_add_node_crashcollector.py

```diff
@@ -0,0 +1,136 @@
+    Add node with OCS label and verify crashcollector created on new node
+
+    Test Procedure:
+    1.Get worker node where mgr pod running [worker-node-x]
+    2.Add worker node with OCS label
+    3.Check ceph status [wait_for_rebalance]
+    4.Drain worker-node-x
+    5.Wait for 3 mon pods to be on running state
+    6.Verify ceph-crashcollector pod running on worker node where "rook-ceph" pods are running.
+    7.Schedule worker-node-x
+    8.Wait for 3 osd pods to be on running state
+    9.Verify ceph-crashcollector pod running on worker node where "rook-ceph" pods are running.
+
+    """
+
+    def test_add_node_crash_collector(self, add_nodes, node_drain_teardown):
+        """
+        Add node with OCS label and verify crashcollector created on new node
+
+        """
+        logger.info("Get Node name where mgr pod running")
+        mgr_pod_nodes = [get_pod_node(pod) for pod in get_mgr_pods()]
+        mgr_pod_node_name = [node.name for node in mgr_pod_nodes]
+
+        logger.info("Add one worker node with OCS label")
+        add_nodes(ocs_nodes=False, node_count=1)
+        ceph_cluster_obj = CephCluster()
+        assert ceph_cluster_obj.wait_for_rebalance(
+            timeout=3600
+        ), "Data re-balance failed to complete"
+
+        drain_nodes(mgr_pod_node_name)
+
+        logging.info("Wait for 3 mon pods to be on running state")
+        pod = OCP(kind=constants.POD, namespace=config.ENV_DATA["cluster_namespace"])
+        assert pod.wait_for_resource(
+            condition="Running",
+            selector=constants.MON_APP_LABEL,
+            resource_count=3,
+            timeout=1400,
+        )
+        logger.info(
+            "Verify rook-ceph-crashcollector pod running on worker node"
+            " where rook-ceph pods are running."
+        )
+        assert sorted(self.get_crashcollector_nodes()) == sorted(
+            self.get_nodes_where_ocs_pods()
+        ), (
+            f"The crashcollector pod does not exist on "
+            f"{self.get_nodes_where_ocs_pods() - self.get_crashcollector_nodes()} "
+            f"even though rook-ceph pods are running on this node"
+        )
+
+        schedule_nodes(mgr_pod_node_name)
+
+        logging.info("Wait for 3 osd pods to be on running state")
+        assert pod.wait_for_resource(
+            condition="Running",
+            selector=constants.OSD_APP_LABEL,
+            resource_count=3,
+            timeout=600,
+        )
+
+        logger.info(
+            "Verify rook-ceph-crashcollector pod running on worker node where rook-ceph pods are running."
+        )
+        assert sorted(self.get_crashcollector_nodes()) == sorted(
+            self.get_nodes_where_ocs_pods()
+        ), (
+            f"The crashcollector pod does not exist on "
+            f"{self.get_nodes_where_ocs_pods() - self.get_crashcollector_nodes()} "
+            f"even though rook-ceph pods are running on this node"
+        )
+
+    def get_crashcollector_nodes(self):
```

Comment:
Can we move it to some common libs place outside the test case

Reply:
done